### DISCUSSION TITLE
Weird bug in http client

### DISCUSSION THREAD

**Discussion by RikudouSage** [symfony/symfony#50152] [Q&A]
### Symfony version(s) affected

6.2.9

### Description

I stumbled upon this error in the logs:

`Uncaught PHP Exception TypeError: "Symfony\Component\HttpClient\Response\CurlResponse::initialize(): Argument #1 ($response) must be of type Symfony\Component\HttpClient\Response\CurlResponse, Symfony\Component\HttpClient\Response\CurlResponse given, called in /var/task/vendor/symfony/http-client/Response/CommonResponseTrait.php on line 41" at /var/task/vendor/symfony/http-client/Response/CommonResponseTrait.php line 132`

### How to reproduce

Sadly I have no idea.

**Comment by ghost** [comment]
Can you provide a reproducer? It's really awkward error.

**Comment by ghost** [comment]
> Can you provide a reproducer? It's really awkward error.

Sorry, I should have read not only the error message but the whole issue :laughing: 

> How to reproduce
Sadly I have no idea.

**Comment by derrabus** [comment]
Let's move this to support until we have a way to reproduce the issue.

**Reply by RikudouSage** [reply]
Makes sense. Unlike the error message. I've never seen something similar where it claims it wants the same type it's getting.

**Comment by HypeMC** [comment]
Which version of PHP are you using? This reminds me of https://github.com/php/php-src/issues/9675#issuecomment-1342558074, but this bug was fixed in https://github.com/php/php-src/pull/10143

**Reply by RikudouSage** [reply]
I'm using 8.2.5. Where can I find which version was it merged into?

**Reply by HypeMC** [reply]
It was fixed in [v8.2.2](https://www.php.net/ChangeLog-8.php#8.2.2) so this is either a different issue, or the fix didn't cover all cases I guess.

**Comment by cybernet** [comment]
still present ( at least for me ) [_51047](https://github.com/symfony/symfony/discussions/51047)
